### PR TITLE
Keyword should exist performance

### PR DESCRIPTION
**Pull request by JFoederer**
In one project I am working on, they ran into an issue where checks were failing because Robot framework was taking up too much time. Analysis showed that the time was taken by calls to the Built in keyword `Keyword should exist`. In that particular situation it could take more than a second for a single call to `Keyword should exist` to return, with an average of 400ms. Anything above 500ms was causing failures.

Further analysis revealed the time to be spend on finding keyword recommendations. Finding these recommendations is a very useful feature when you misspell a keyword during test case development. However, when explicitly checking for a keyword's existence at runtime, the boolean answer suffices. In fact, in the current implementation the recommendations are disposed and not included in the reported failure.

This pull request makes the recommendation finding optional. By default it is still enabled, but when executing `Keyword should exist`, it is disabled for aforementioned performance reasons.

### CODE REVIEW

**Comment by JFoederer** [comment]
Note that the tests that were added cover existing functionality, previously not covered. External behavior did not change and no documentation updates are needed.

**Comment by pekkaklarck** [comment]
Makes sense. Quick comments:
- Could you submit an issue about this for tracking purposes?
- There ought to be tests for including recommendations already so there shouldn't be need to add new for that feature. I think they are in a separate suite like `keyword_recommendations.robot`.
- I'm currently working with a rather big change to `get_runner` method this PR modifies related to #4454. There will be a conflict but it ought to be pretty easy to fix. Hopefully that's not too big a problem.

**Comment by pekkaklarck** [comment]
I have committed all changes I mentioned above and apparently they didn't touch exactly this code as there are no conflicts. Tests could be changed a bit, though, but I add separate comments about that. Having an issue for tracking purposes would be good too.

**Review by pekkaklarck — APPROVED** [review]
Looks great but tests could be enhanced a bit.

File: atest/robot/standard_libraries/builtin/keyword_should_exist.robot [hunk 1]
```diff
@@ -31,6 +31,9 @@ Keyword does not exist
 Keyword does not exist with custom message
     Check Test Case  ${TESTNAME}
 
+Misspelled keyword does not exist, no recommendations
```
[the remark this answers is not in the document]
Comment (JFoederer) [reply atest/robot/standard_libraries/builtin/keyword_should_exist.robot:34]: I like the requirement style phrasing better, but felt the current one matched the style of the other tests in the file better.
How about 

> Recommendations not shown if keyword does not exist

File: atest/robot/keywords/keyword_not_found.robot [hunk 2]
```diff
@@ -21,5 +21,8 @@ Non Existing Implicit In User Keyword
 Non Existing Explicit In User Keyword
     Check Test Case    ${TEST NAME}
 
+Misspelled keyword yields recommendations
+    Check Test Case    ${TEST NAME}
```
[the remark this answers is not in the document]
Comment (JFoederer) [reply atest/robot/keywords/keyword_not_found.robot:25]: I hadn't spotted that one. That does indeed make this one redundant. I will remove it.

File: atest/testdata/standard_libraries/builtin/keyword_should_exist.robot [hunk 3]
```diff
@@ -45,6 +45,10 @@ Keyword does not exist with custom message
     [Documentation]  FAIL Custom message
     Non Existing  Custom message
 
+Misspelled keyword does not exist, no recommendations
+    [Documentation]  FAIL No keyword with name 'should be eQQual' found.
+    should be eQQual  No keyword with name 'should be eQQual' found.
```
[the remark this answers is not in the document]
Comment (JFoederer) [reply atest/testdata/standard_libraries/builtin/keyword_should_exist.robot:50]: Good point. I somehow must have thought that was needed, but it obviously isn't. I will remove the argument.

[the remark this answers is not in the document]
Comment (JFoederer) [reply atest/testdata/standard_libraries/builtin/keyword_should_exist.robot:50]: Do you think it is clear enough that `Should be eQQual` would, when used as a keyword, yield recommendations? The counter check that proved that it did has now been removed.

[the remark this answers is not in the document]
Comment (pekkaklarck) [reply atest/testdata/standard_libraries/builtin/keyword_should_exist.robot]: I think it's pretty clear if the test name is explicit. A comment could be added to explain it in more detail if you think it helps.

The counter example would be in another file anyway so it wouldn't be seen here.

File: atest/robot/standard_libraries/builtin/keyword_should_exist.robot (hunk 1 above)
[the remark this answers is not in the document]
Comment (pekkaklarck) [reply atest/robot/standard_libraries/builtin/keyword_should_exist.robot]: That looks good to me.

**Comment by JFoederer** [comment]
Kiitos @pekkaklarck for getting this in so quickly. Greatly appreciated!

**Pull request merged**